### DISCUSSION TITLE
Track Number of Pending items?

### DISCUSSION THREAD

**Discussion by Tekpakma** [rpldy/react-uploady#505] [Q&A]
Hi, so I was fiddling around for quite a while now, but I couldn't figure it out myself.

What I want at the end is the number of Pending Items,  which I would use for some UI changes. What I considered but didn't liked was tracking them by myself.

What looked promising was getPending() from useUploader, 
but couldn't import useUploader which was weird, so I also didn't tried that out.

Would appreciate some input!.

**Comment by yoavniran** [comment]
So a few things here:

1. getPending isnt an actual method. I think its leftover (cant remember now) of no longer available API method. So you cant rely on it. The only ref left is in the d.ts file and I will clean that shortly

2. Its strange that you couldnt import useUploader, although you shouldn't need it. What didnt work?

3. When you mean track number of pending, what do you mean by that? just actual pending, as in files that are not auto-uploaded, or files that havent started to upload as well (because there are already uploading items/batches)? 

In any case, the best way would be to track that yourself, that is intentionally not something that Uploady provides as an API. Instead, you have all the events (and hooks) to help you build that (custom) logic on your own.

**Reply by Tekpakma** [reply]
To 2. Probably some caching problems with my environment setup then, might need to reinstall node modules
3. Yes Files that are not auto-uploaded. 

I guess I'll use Events then to achieve this. Thank you!
